Feat/api integration

AhmedRamadanAR opened the pull request:
Api Integration and dealing with DOM

AhmedRamadanAR committed:
feat(search-fetch): implement search by city feature

Mahmoud-Hanyy committed:
Styling Home page is done

Mahmoud-Hanyy committed:
Merge pull request #15 from Mahmoud-Hanyy/feat/CSS-right-side

Styling Home page is done

AhmedRamadanAR committed:
feat:(Api-Integration)

AhmedRamadanAR committed:
feat(API-integraion): integrate API and DOM manipulation

AhmedRamadanAR committed:
feat(search-fetch): add api integration and DOM

Added the following JavaScript features to the Weather App:
  - Integrated the openweathermap.org api
  - Updated the UI based on user interactivity using DOM features
  - Fetching weather app by searching the city name

Ref #10

AAEmara committed:
chore: delete unnecessary files

Deleted unnecessary vscode files.

AAEmara commented on scripts/script.js:
```diff
@@ -0,0 +1,155 @@
+const apiKey= '9853aca6055d1f5c122be191411d006b';
```
```suggestion
const apiKey= process.env.API_Key;
```

Please use an env file to include the API_Key.
Also, don't forget to invalidate this API Key.

AAEmara reviewed (changes requested):
Please, make the JavaScript reference an `env` file that contains the API_KEY and invalidate the the old API Key.

For the code style, I will fix it on a later commit, and share a standard lint.

AhmedRamadanAR committed:
fix(search-fetch): remove api key and add it in separated  file

Modified the following in the JavaScript file for Weather App:
  - Used an json file to store the API Key.
  -fix forecast's icons
  - Linked the API_KEY value in JavaScript

Partially-Addresses #10

AhmedRamadanAR committed:
Merge branch 'feat/api_integration' of https://github.com/Mahmoud-Hanyy/WindyX into feat/api_integration

AAEmara commented on scripts/script.js:
```diff
@@ -0,0 +1,169 @@
+let apikey;
```
```suggestion
let apiKey;
```

AAEmara reviewed (changes requested):
Great job @AhmedRamadanAR, Please fix the variable name `apikey` typo mentioned.
The code style will be fixed from my side.

Thank you.

AAEmara committed:
Merge branch 'main' into feat/api_integration

AhmedRamadanAR committed:
Merge branch 'feat/api_integration' of https://github.com/Mahmoud-Hanyy/WindyX into feat/api_integration

AhmedRamadanAR committed:
refactor: rename apikey to apiKey

#10
